app/services/google_parse_service.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class GoogleParseService
```

Comment:
I think we can refactor this service into PORO (Plain Old Ruby Object) because other places rely on keys of the results, such as the `update_data` method in the keyword model. This can cause an issue when any key gets changed, it will break others' functions. 💭

The following example if `ads_top_count` get renamed, it going to raise an error during the compile time not at the runtime so we got extra safety here. 😄
```ruby
class GoogleResult
  def initialize(html:)
    @html = html
    @document = Nokogiri::HTML(html)
  end

  def ads_top_count
    @ads_top_count ||= document.css('#tads > div').count
  end
end
```

Reply:
Nice suggestion!
Fixed in [`5f004ca` (#47)](https://github.com/suho/google-search-ruby/pull/47/commits/5f004cae025caf7b86ce910239bd5478850fb86a)